diff --git a/node_modules/facebook-chat-api/DOCS.md b/node_modules/facebook-chat-api/DOCS.md
index 9a589c8..e85721a 100644
--- a/node_modules/facebook-chat-api/DOCS.md
+++ b/node_modules/facebook-chat-api/DOCS.md
@@ -1626,6 +1626,7 @@ __Arguments__
     - `userAgent`: (Default `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.3.18 (KHTML, like Gecko) Version/8.0.3 Safari/600.3.18`) The desired simulated User Agent.
 	- `autoMarkDelivery`: (Default `true`) Will automatically mark new messages as delivered. See [api.markAsDelivered](#markAsDelivered).
 	- `autoMarkRead`: (Default `false`) Will automatically mark new messages as read/seen. See [api.markAsRead](#markAsRead).
+	- `online`: (Default `true`) Set account's online state.
 
 __Example__
 
diff --git a/node_modules/facebook-chat-api/index.js b/node_modules/facebook-chat-api/index.js
index 0ad1cd5..94d4369 100644
--- a/node_modules/facebook-chat-api/index.js
+++ b/node_modules/facebook-chat-api/index.js
@@ -10,6 +10,9 @@ log.maxRecordSize = defaultLogRecordSize;
 function setOptions(globalOptions, options) {
   Object.keys(options).map(function(key) {
     switch (key) {
+      case 'online':
+        globalOptions.online = options.online;
+        break;
       case 'logLevel':
         log.level = options.logLevel;
         globalOptions.logLevel = options.logLevel;
@@ -394,7 +397,7 @@ function loginHelper(appState, email, password, globalOptions, callback) {
         });
     })
     .then(function(resData) {
-      if (resData.t !== 'lb') throw {error: "Bad response from pull 1"};
+      if (resData.t !== 'lb') throw {error: "Bad response from base pull"};
 
       var form = {
         channel : 'p_' + ctx.userID,
@@ -410,23 +413,35 @@ function loginHelper(appState, email, password, globalOptions, callback) {
         sticky_token: resData.lb_info.sticky,
         sticky_pool: resData.lb_info.pool,
       };
-
-      log.info("login", "Request to pull 2");
-      return utils
-        .get("https://0-edge-chat.facebook.com/pull", ctx.jar, form, globalOptions)
-        .then(utils.saveCookies(ctx.jar));
-    })
-    .then(function() {
-      var form = {
-        'client' : 'mercury',
-        'folders[0]': 'inbox',
-        'last_action_timestamp' : '0'
+      var lastOnlineTime = new Date();
+      var ping = function () {
+        log.info("login", "Request to pull (ping)");
+        utils.get("https://0-edge-chat.facebook.com/pull", ctx.jar, form, globalOptions)
+          .then(utils.saveCookies(ctx.jar))
+          .then(function () { 
+            if (globalOptions.online) {
+              lastOnlineTime = new Date();
+            }
+            form = {
+              channel: 'p_' + ctx.userID,
+              seq: 0,
+              partition: -2,
+              clientid: ctx.clientID,
+              viewer_uid: ctx.userID,
+              uid: ctx.userID,
+              idle: Math.floor((new Date().getTime() - lastOnlineTime.getTime()) / 1000),
+              cap: 8,
+              msgs_recv: 0,
+              sticky_token: resData.lb_info.sticky,
+              sticky_pool: resData.lb_info.pool,
+            };
+            if (globalOptions.online) {
+              form.state = 'active';
+            }
+            setTimeout(ping);
+          });
       };
-      log.info("login", "Request to thread_sync");
-
-      return defaultFuncs
-        .post("https://www.facebook.com/ajax/mercury/thread_sync.php", ctx.jar, form, globalOptions)
-        .then(utils.saveCookies(ctx.jar));
+      setTimeout(ping);
     });
 
   // given a pageID we log in as a page
@@ -471,7 +486,8 @@ function login(loginData, options, callback) {
     autoMarkDelivery: true,
     autoMarkRead: false,
     logRecordSize: defaultLogRecordSize,
-    userAgent: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.3.18 (KHTML, like Gecko) Version/8.0.3 Safari/600.3.18"
+    userAgent: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.3.18 (KHTML, like Gecko) Version/8.0.3 Safari/600.3.18",
+    online: true
   };
 
   setOptions(globalOptions, options);
